src/scripts/main.js

```diff
@@ -1,3 +1,324 @@
 'use strict';
 
-// write code here
+// TABLE_VARIABLES
+const table = document.querySelector('table');
+const tHead = table.tHead;
+const tBody = table.tBodies[0];
+
+// #region SORTING
+// SORTING FUNCTIONS
+const valueConversionFunctions = {
+  age: parseInt,
+  salary: window.convertSalartyToNumber,
+};
+
+const findSuitableSortingFunction = (
+  index,
+  order,
+  transformFunc = (x) => x,
+) => {
+  return (a, b) => {
+    const valueA = transformFunc(window.getValueByCellIndex(a, index));
+    const valueB = transformFunc(window.getValueByCellIndex(b, index));
+
+    if (typeof valueA === 'string' && typeof valueB === 'string') {
+      return order * valueA.localeCompare(valueB);
+    }
+
+    return order * (valueA - valueB);
+  };
+};
+
+const getSortingFunction = (index, colHeader, ascOrder) => {
+  const transformFunc = valueConversionFunctions[colHeader];
+
+  return ascOrder
+    ? findSuitableSortingFunction(index, 1, transformFunc)
+    : findSuitableSortingFunction(index, -1, transformFunc);
+};
+
+// TOGGLE ASC/DESC ORDER LOGIC
+const ascDescOrderUtil = {
+  element: undefined,
+  ascOrder: undefined, // true -> asc, false -> desc
+
+  clickIsMade: (target) => {
+    if (target === ascDescOrderUtil.element) {
+      ascDescOrderUtil.ascOrder = !ascDescOrderUtil.ascOrder;
+    } else {
+      ascDescOrderUtil.element = target;
+      ascDescOrderUtil.ascOrder = true;
+    }
+  },
+  getValue: () => ascDescOrderUtil.ascOrder,
+};
+
+// HANDLING ASC/DESC CLICK EVENT
+const handleSortingEvent = (e) => {
+  const target = e.target.closest('th');
+
+  if (target) {
+    ascDescOrderUtil.clickIsMade(target);
+
+    const index = target.cellIndex;
+    const colHeader = target.textContent.toLowerCase();
+    const orderDirection = ascDescOrderUtil.getValue();
+
+    Array.from(tBody.rows)
+      .sort(getSortingFunction(index, colHeader, orderDirection))
+      .forEach((row) => tBody.append(row));
+  }
+};
+
+tHead.addEventListener('click', handleSortingEvent);
+// #endregion
+
+// #region SELECT_ROW
+// HANDLING ROW SELECT EVENT
+const deselectRows = () => {
+  for (const row of table.rows) {
```

Comment:
In the `deselectRows` function, you are using a loop to remove the 'active' class from all rows. This could be inefficient if there are a lot of rows. Consider keeping track of the currently active row and only remove the 'active' class from that row when needed.